com.unity.shadergraph/Editor/ShaderFoundry/Utilities/SamplerStateAttribute.cs

```diff
@@ -0,0 +1,219 @@
+
+namespace UnityEditor.ShaderFoundry
+{
+    internal class SamplerStateAttribute
+    {
+        public enum FilterModeEnum
+        {
+            Point,
+            Linear,
+            Trilinear
+        }
+        public enum WrapModeParameterStates
+        {
+            Clamp, ClampU, ClampV, ClampW,
+            Repeat, RepeatU, RepeatV, RepeatW,
+            Mirror, MirrorU, MirrorV, MirrorW,
+            MirrorOnce, MirrorOnceU, MirrorOnceV, MirrorOnceW
+        }
+        public enum WrapModeEnum
+        {
+            Clamp, Repeat, Mirror, MirrorOnce
+        }
+
+        internal const char WrapModeDelimeter = ',';
+        internal const string AttributeName = "SamplerState";
+        internal const string FilterModeParamName = "filterMode";
+        internal const string WrapModeParamName = "wrapMode";
+        internal const string DepthCompareParamName = "depthCompare";
+        internal const string AnisotropicLevelParamName = "anisotropicLevel";
+
+        public FilterModeEnum FilterMode = FilterModeEnum.Linear;
+        public WrapModeEnum WrapModeU = WrapModeEnum.Repeat;
+        public WrapModeEnum WrapModeV = WrapModeEnum.Repeat;
+        public WrapModeEnum WrapModeW = WrapModeEnum.Repeat;
+        public bool DepthCompare = false;
+        public int AnisotropicLevel = 0;
+
+        internal static SamplerStateAttribute FindFirst(IEnumerable<ShaderAttribute> attributes)
+        {
+            var attribute = attributes.FindFirst(AttributeName);
+            if (attribute.IsValid)
+                return TryParse(attribute);
+            return null;
+        }
+
+        static AttributeParsing.SignatureDescription<SamplerStateAttribute> AttributeSignature = new AttributeParsing.SignatureDescription<SamplerStateAttribute>()
+        {
+            ParameterDescriptions = new List<AttributeParsing.ParameterDescription<SamplerStateAttribute>>
+            {
+                new AttributeParsing.ParameterDescription<SamplerStateAttribute>(FilterModeParamName, (param, index, target) => AttributeParsing.ParseEnum<FilterModeEnum>(param, index, ref target.FilterMode)),
+                new AttributeParsing.ParameterDescription<SamplerStateAttribute>(WrapModeParamName, (param, index, target) => ParseWrapMode(param, index, ref target)),
+                new AttributeParsing.ParameterDescription<SamplerStateAttribute>(DepthCompareParamName, (param, index, target) => AttributeParsing.ParseBool(param, index, ref target.DepthCompare)),
+                new AttributeParsing.ParameterDescription<SamplerStateAttribute>(AnisotropicLevelParamName, (param, index, target) => ParseAnisotropicLevel(param, index, ref target.AnisotropicLevel)),
+            },
+        };
+
+        static void ParseWrapMode(ShaderAttributeParam attributeParam, int parameterIndex, ref SamplerStateAttribute target)
+        {
+            var tokens = attributeParam.Value.Split(WrapModeDelimeter);
+            foreach (var token in tokens)
```

Comment:
So it's order dependent and last one wins, right?
If you go with "ClampU, Repeat" it will do repeat for all three.

Reply:
I think we should warn about overwriting the state. It's either redundant and it's better to fix it or it's incorrect and needs to be fixed.

Reply:
Updated to warn now. Added tests to validate the warning and the override behavior.
To note, currently this works similar to the underlying uniform name, so if you specify:
`ClampU,Repeat" then the result is "Repeat_ClampU" not "Repeat", that is specifying the full-channel value doesn't override any single channel values.